cfrontend/PrintClight.ml

```diff
@@ -266,19 +272,35 @@ let print_globdef p (id, gd) =
   | AST.Gfun f -> print_fundef p id f
   | AST.Gvar v -> print_globvar p id v  (* from PrintCsyntax *)
 
+let collect_internal_fct defs =
+  let add_decl (id,gd) l =
+    match gd with
+    | AST.Gfun (Internal f) -> (id,f)::l
+    |  _ -> l in
+  List.fold_right add_decl defs []
+
+let print_fundecl p (id,fct) =
+  fprintf p "%s;@ @ " (name_cdecl (extern_atom id) (Clight.type_of_function fct))
+
```

Comment:
There is no real need for building a list of what you're going to print, then print by iterating over this list.  Why not iterate over the whole list of definitions and choose what you're going to print?

Reply:
This could indeed be done on-the-fly.